### PR TITLE
Print error message when passing a non-existant socket path in the PAB

### LINKED ISSUE #287: PAB does not report invalid node socket path
### Summary

The PAB silently fails to handle blockchain effects if it cannot connect to the node socket in its configuration.

### Steps to reproduce the behavior

1. Edit the configuration file for the PAB and set `pscSocketPath` to any path that does not exist.
2. Start the PAB.

### Actual Result

Observe that no error message is reported, but the PAB runs without a connection to the node, and hence never notifies clients of blockchain effects..

### Expected Result

The PAB should fail because of this fatal misconfiguration and report that it could not connect to the node.

### Describe the approach you would take to fix this

Add an error handler and logger to https://github.com/input-output-hk/plutus-apps/blob/ea1bfc6a49ee731c67ada3bfb326ee798001701a/plutus-pab/src/Plutus/PAB/Core/ContractInstance/BlockchainEnv.hs#L58.

### System info

any

### PR DESCRIPTION
**Pull request by koslambrou**
Fixes #287 

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
    - [x] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested

### CODE REVIEW

File: plutus-pab/src/Plutus/PAB/Run/Cli.hs [hunk 1]
```diff
@@ -128,53 +131,64 @@ runConfigCommand _ ConfigCommandArgs{ccaTrace, ccaPABConfig = Config {nodeServer
...
-            Beam.runBeamStoreAction connection (LM.convertLog LM.PABMsg ccaTrace)
-            $ interpret (LM.handleLogMsgTrace ccaTrace)
-            $ do
-                cIds   <- Map.toList <$> Contract.getActiveContracts @(Builtin a)
-                forM cIds $ \(cid, args) -> do
-                    s <- Contract.getState @(Builtin a) cid
-                    let priorContract :: (SomeBuiltinState a, Wallet.ContractInstanceId, ContractActivationArgs a)
-                        priorContract = (s, cid, args)
-                    pure priorContract
+runConfigCommand
```
Comment (koslambrou) [inline plutus-pab/src/Plutus/PAB/Run/Cli.hs:150]: Changes in this function are only formatting

**Comment by koslambrou** [comment]
@bwbush This works for you as well? Since we print the information on the tip of the cardano using the socket, it will depend wheter the socket file doesn't exist.

**Review by bwbush — CHANGES_REQUESTED** [review]
The changes are okay, but they don't fix the problem.

When I run `plutus-pab-executables` with a valid socket, I get
```
[nix-shell:plutus-apps.PR323/plutus-pab-executables]$ git log -n 1 --oneline
e72e60adf (HEAD -> kll-pab-invalid-socket-path, origin/kll-pab-invalid-socket-path) Print error message when passing a non-existant socket path in the PAB

[nix-shell:plutus-apps.PR323/plutus-pab-executables]$ cabal run exe:plutus-pab-examples -- webserver --config marlowe-pab.yaml
Up to date
[pab:Info:19] [2022-02-23 18:01:41.21 UTC] Restoring PAB state ...
[pab:Info:19] [2022-02-23 18:01:41.21 UTC] No contract instance were restored in the PAB state.
[pab:Info:19] [2022-02-23 18:01:41.21 UTC] Starting PAB backend server on port 9080
Current block: 95. Current slot: 127678
Current block: 97. Current slot: 161191
Current block: 252. Current slot: 363718
^C
```
but when I remove the socket file, I get continuing silence and no error message:
```
[nix-shell:plutus-apps.PR323/plutus-pab-executables]$ rm node.socket 

[nix-shell:plutus-apps.PR323/plutus-pab-executables]$ cabal run exe:plutus-pab-examples -- webserver --config marlowe-pab.yaml
Up to date
[pab:Info:19] [2022-02-23 18:01:51.92 UTC] Restoring PAB state ...
[pab:Info:19] [2022-02-23 18:01:51.92 UTC] No contract instance were restored in the PAB state.
[pab:Info:19] [2022-02-23 18:01:51.92 UTC] Starting PAB backend server on port 9080
^C
```

This needs to pass several test cases:
1. *Success:* the socket file is connected to a running node.
2. *Failure:* the socket file exists but is not connected to a running node.
3. *Failure:* the socket file does not exist.

**Comment by koslambrou** [comment]
@bwbush Oops, I put the code in the wrong place. Should be fixed now.
It now correctly outputs an error when the socket file doesn't exist in AlonzoMode.

**Review by bwbush — APPROVED** [review]
Works perfectly now!
```
[nix-shell:plutus-apps.PR323/plutus-pab-executables]$ ln -s ../../networks/private/node.socket 
[nix-shell:plutus-apps.PR323/plutus-pab-executables]$ cabal run exe:plutus-pab-examples -- webserver --config ../../networks/private/marlowe-pab.yaml
[pab:Info:19] [2022-02-24 16:26:20.73 UTC] Connecting to Alonzo node with config:
Socket: node.socket
Network Id: Testnet (NetworkMagic {unNetworkMagic = 1564})
Port: 80
Security Param: 2160
The tip of the local node: SlotNo 790305
[pab:Info:19] [2022-02-24 16:26:20.81 UTC] Restoring PAB state ...
[pab:Info:19] [2022-02-24 16:26:20.81 UTC] No contract instance were restored in the PAB state.
[pab:Info:19] [2022-02-24 16:26:20.81 UTC] Starting PAB backend server on port 9080
Current block: 72. Current slot: 105761
^C

[nix-shell:../../plutus-apps.PR323/plutus-pab-executables]$ rm node.socket
[nix-shell:plutus-apps.PR323/plutus-pab-executables]$ cabal run exe:plutus-pab-examples -- webserver --config ../../networks/private/marlowe-pab.yaml
plutus-pab-examples: Network.Socket.connect: <socket: 74>: does not exist (No such file or directory)

[nix-shell:../../plutus-apps.PR323/plutus-pab-executables]$ rm node.socket
[nix-shell:plutus-apps.PR323/plutus-pab-executables]$ mkfifo node.socket
[nix-shell:plutus-apps.PR323/plutus-pab-executables]$ cabal run exe:plutus-pab-examples -- webserver --config ../../networks/private/marlowe-pab.yaml
plutus-pab-examples: Network.Socket.connect: <socket: 74>: does not exist (Connection refused)
```

**Pull request merged**